File: backend/src/budget/data/model/Budget.ts
import mongoose from "mongoose";
import pagination from "mongoose-paginate-v2"

export interface BudgetDoc extends mongoose.Document{
    type: string,
    accountID: string,
    goalAmount: number,
    utilized: number,
    endDate: string,
    description: string,

}

export interface BudgetModel extends mongoose.PaginateModel<BudgetDoc>{}

const BudgetSchema = new mongoose.Schema({
    type: {type: String, required: true},
    accountID: {type: String, required: true},
    goalAmount: {type: Number, required: true},
    utilized: {type: String},
    endDate: {type: String, required: true},
    description: {type: String},
})

BudgetSchema.plugin(pagination)
export default BudgetSchema

File: tilla_team/lib/domain/budget/budget.dart
class Budget {
  final String type;
  final String accountID;
  final String utilized;
  final String goalAmount;
  final String endDate;
  final String description;

  Budget(
      {required this.type,
      required this.accountID,
      required this.utilized,
      required this.goalAmount,
      required this.endDate,
      required this.description});

  factory Budget.fromJson(Map<String, dynamic> json) {
    return Budget(
      type: json['type'],
      accountID: json['accountID'],
      utilized: json['utilized'],
      goalAmount: json['goalAmount'],
      endDate: json['endDate'],
      description: json['description'],
    );
  }

  Map<String, dynamic> toJson() => {
        'type': type,
        'accountID': accountID,
        'utilized': utilized,
        'goalAmount': goalAmount,
        'endDate': endDate,
        'description': description,
      };
}
